### PR TITLE
Make rename-mapping to support assigning format

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

File: test/TestProjects/MgmtRenameRules/readme.md [hunk 1]
```diff
@@ -46,6 +45,6 @@ rename-mapping:
   SshPublicKeyResource: SshPublicKey
   LogAnalyticsOperationResult: LogAnalytics
   RollingUpgradeStatusInfo: VirtualMachineScaleSetRollingUpgrade
-  UpgradeOperationHistoricalStatusInfo.type: ResourceType
+  UpgradeOperationHistoricalStatusInfo.type: ResourceType|resource-type
```
Comment (fengzhou-msft) [inline test/TestProjects/MgmtRenameRules/readme.md:48]: If I want to add a format for a property, but not rename it. Do I have to put `Type|resource-type`? Do you think it adds value if we support `|resource-type` to do format only?

Comment (ArcturusZhang) [reply test/TestProjects/MgmtRenameRules/readme.md]: I explained this scenario in the document

File: docs/mgmt/polishing.md [hunk 2]
```diff
@@ -183,6 +185,33 @@ public partial class Model
 }
 ```
 
+### Change the format of a property
+
+To assign a new format to a property, you could use this syntax:
+```yaml
+rename-rules:
```
Comment (ArthurMa1978) [inline docs/mgmt/polishing.md:192]: rename-mapping

File: docs/mgmt/polishing.md [hunk 3]
```diff
@@ -183,6 +185,33 @@ public partial class Model
...
+-    public string OldProperty { get; set; }
++    public ResourceType? NewProperty { get; set; }
+
+    /* other things inside the class */
+}
+```
+
+If only the type of this property needs change, you could omit its new name, like
+```yaml
+rename-rules:
```
Comment (ArthurMa1978) [inline docs/mgmt/polishing.md:210]: rename-mapping

File: docs/mgmt/polishing.md [hunk 4]
```diff
@@ -183,6 +185,33 @@ public partial class Model
...
+  Model.oldProperty: -|resource-type
+```
+Please note that the dash and slash `-|` here are mandatory as a placeholder for the property name. The generator uses this symbol to separate the part for property name and its format.
```
Comment (ArthurMa1978) [inline docs/mgmt/polishing.md:213]: what's the scenario `name | - `?

**Review by ArthurMa1978 — APPROVED** [review]

File: docs/mgmt/polishing.md (hunk 4 above)
Comment (ArcturusZhang) [reply docs/mgmt/polishing.md:213]: When you only want to change its name without changing its format which is equivalent to just name without `|-`

**Pull request merged**